src/exits/tasks.py

```diff
@@ -48,6 +52,39 @@ async def update_exit_signatures(
         )
 
 
+async def _check_majority_oracles_synced(oracles: Oracles, update_block: BlockNumber) -> bool:
+    threshold = oracles.validators_threshold
+    pending = {
+        asyncio.create_task(_fetch_last_update_block_replicas(replicas))
+        for replicas in oracles.endpoints
+    }
+    while pending:
+        done, pending = await asyncio.wait(pending, return_when=asyncio.FIRST_COMPLETED)
+        for task in done:
+            block_number = task.result()
+            if not block_number or block_number < update_block:
+                continue
+            threshold -= 1
+            if threshold <= 0:
+                for task in pending:
+                    task.cancel()
+                return True
+    return False
+
+
+async def _fetch_last_update_block_replicas(replicas: list[str]) -> BlockNumber | None:
+    results = await asyncio.gather(
+        *[_fetch_exit_signature_block(endpoint) for endpoint in replicas], return_exceptions=True
+    )
+    blocks = []
+    for res in results:
+        if not isinstance(res, Exception) and res is not None:
+            blocks.append(res)
+    if blocks:
+        return max(blocks)
```

Comment:
I think it should be `min`. That's the worst endpoint we will hit when receiving approvals.

Reply:
no, if one replica stops, all will be down for us.